### PR TITLE
Exile's Reach Taming the Wilds

### PR DESCRIPTION
**Pull request by ghost**
**Changes proposed:**

-  Add Exile's Reach first Hunter training quest.

**Issues addressed:**

Fixed what I can any remaining issues is beyond my capibilities.

**Tests performed:**

Compiles and can be completed. Still issues to work on.

**Known issues and TODO list:** (add/remove lines as needed)

- [ 1 ] On offical you can't dismiss pet.
- [ 2 ] On official if you drop quest and pick it back up Quest objectives remain complete. "Hacked for now research is needed"
- [ 3 ] Pet is very very small....

### CODE REVIEW

File: src/server/scripts/ExilesReach/zone_exiles_reach.cpp [hunk 1]
```diff
@@ -3805,6 +3805,290 @@ class spell_scene_linger_northbound: public SpellScript
...
+    ObjectGuid _playerGUID;
+};
+
+CreatureAI* HuxsworthBriarpatchSelector(Creature* creature)
+{
+    if (creature->IsPrivateObject())
+    {
+        if (Player* privateObjectOwner = ObjectAccessor::GetPlayer(*creature, creature->GetPrivateObjectOwner()))
+        {
+            if ((privateObjectOwner->GetQuestStatus(59342) == QUEST_STATUS_INCOMPLETE))
```
Comment (mdX7) [inline src/server/scripts/ExilesReach/zone_exiles_reach.cpp:4049]: magic number

File: src/server/scripts/ExilesReach/zone_exiles_reach.cpp [hunk 2]
```diff
@@ -3805,6 +3805,290 @@ class spell_scene_linger_northbound: public SpellScript
...
+    return new NullCreatureAI(creature);
+};
+
+CreatureAI* DawntrackerBriarpatchSelector(Creature* creature)
+{
+    if (creature->IsPrivateObject())
+    {
+        if (Player* privateObjectOwner = ObjectAccessor::GetPlayer(*creature, creature->GetPrivateObjectOwner()))
+        {
+            if ((privateObjectOwner->GetQuestStatus(59937) == QUEST_STATUS_INCOMPLETE))
```
Comment (mdX7) [inline src/server/scripts/ExilesReach/zone_exiles_reach.cpp:4063]: magic number

File: src/server/scripts/ExilesReach/zone_exiles_reach.cpp [hunk 3]
```diff
@@ -3805,6 +3805,290 @@ class spell_scene_linger_northbound: public SpellScript
...
+        {
+            switch (eventId)
+            {
+            case EVENT_ME_TURN_TO_PLAYER:
+                if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                {
+                    me->SetFacingToObject(player);
+                    me->SetEmoteState(EMOTE_STATE_TALK);
+                    Conversation* conversation = Conversation::CreateConversation(CONVERSATION_HORDE_TRAINER, player, *player, player->GetGUID(), nullptr, false);
+                    conversation->AddActor(CONVERSATION_HORDE_ACTOR, 0, me->GetGUID());
```
Comment (mdX7) [inline src/server/scripts/ExilesReach/zone_exiles_reach.cpp:4007]: missing nullcheck for conversation (possible crash)

File: src/server/scripts/ExilesReach/zone_exiles_reach.cpp (hunk 1 above)
Comment (malcrom) [reply src/server/scripts/ExilesReach/zone_exiles_reach.cpp]: Done

File: src/server/scripts/ExilesReach/zone_exiles_reach.cpp (hunk 2 above)
Comment (malcrom) [reply src/server/scripts/ExilesReach/zone_exiles_reach.cpp]: Done

File: src/server/scripts/ExilesReach/zone_exiles_reach.cpp (hunk 3 above)
Comment (malcrom) [reply src/server/scripts/ExilesReach/zone_exiles_reach.cpp]: Done

**Pull request merged**